runtime/lua/vim/health.lua

```diff
@@ -307,72 +295,70 @@ local function shellify(cmd)
   return table.concat(escaped, ' ')
 end
 
+--- @private
 function M.cmd_ok(cmd)
   local out = vim.fn.system(cmd)
   return vim.v.shell_error == 0, out
 end
 
---- Run a system command and timeout after 30 seconds.
+--- @private
+---
+--- Run a system command and timeout after {timeout} seconds, and return stdout as string.
+--- When the command fails (non-zero exit code), throw an error with stdout/stderr output (unless
+--- ignore_error is set).
 ---
---- @param cmd table List of command arguments to execute
---- @param args ?table Optional arguments:
----                   - stdin (string): Data to write to the job's stdin
----                   - stderr (boolean): Append stderr to stdout
----                   - ignore_error (boolean): If true, ignore error output
----                   - timeout (number): Number of seconds to wait before timing out (default 30)
-function M.system(cmd, args)
-  args = args or {}
-  local stdin = args.stdin or ''
-  local stderr = vim.F.if_nil(args.stderr, false)
-  local ignore_error = vim.F.if_nil(args.ignore_error, false)
-
-  local shell_error_code = 0
-  local opts = {
-    add_stderr_to_output = stderr,
-    output = '',
-    stderr = '',
-    on_stdout = system_handler,
-    on_stderr = system_handler,
-    on_exit = function(_, data)
-      shell_error_code = data
-    end,
-  }
-  local jobid = vim.fn.jobstart(cmd, opts)
-
-  if jobid < 1 then
-    local message =
-      string.format('Command error (job=%d): %s (in %s)', jobid, shellify(cmd), vim.loop.cwd())
-    error(message)
-    return opts.output, 1
+--- @param cmd string[] Command arguments to execute
+--- @param opts? table Optional arguments:
+---               - stderr (boolean): Append stderr to stdout
+---               - ignore_error (boolean): If true, don't raise but return error as string.
+---               - timeout (number): Number of seconds to wait before timing out (default 30)
+--- @return string stdout output, also stderr if { stderr = true }
+--- @return integer exit code
+function M.system(cmd, opts)
+  opts = opts or {}
+  local stderr = vim.F.if_nil(opts.stderr, false)
+  local ignore_error = vim.F.if_nil(opts.ignore_error, false)
+  local timeout = vim.F.if_nil(opts.timeout, 30) * 1000 ---@type integer
+
+  local ok, p = pcall(function() -- catch ENOENT errors (command not found)
+    return vim.system(cmd, { timeout = timeout }):wait()
+  end)
+  if not ok then
+    local err = p --[[ @as string ]]
+    if ignore_error then
+      return ('Command error: %s\n (in %s)\n'):format(shellify(cmd), vim.loop.cwd()) .. err, 1
```

Comment:
Do we actually need this line? From what I can tell it's not used appropriately from any of the callers.

Reply:
For now it's not used, but it would be a more consistent behavior to not throw when `ignore_error = true`.